app.go

```diff
@@ -881,16 +881,37 @@ func (e *Error) Error() string {
 	return e.Message
 }
 
-// NewError creates a new Error instance with an optional message
-func NewError(code int, message ...string) *Error {
-	err := &Error{
-		Code:    code,
-		Message: utils.StatusMessage(code),
-	}
-	if len(message) > 0 {
-		err.Message = message[0]
+// NewError creates a new Error instance with an optional message.
+// Additional arguments are formatted using fmt.Sprintf when provided.
+// If the first argument in the message slice is not a string, the function
+// falls back to using fmt.Sprint on the first element to generate the message.
+func NewError(code int, message ...any) *Error {
+	var msg string
+
+	switch len(message) {
+	case 0:
+		// nothing to override
+		msg = utils.StatusMessage(code)
+
+	case 1:
+		// One argument → treat it like fmt.Sprint(arg)
+		if s, ok := message[0].(string); ok {
```

Comment:
This check is probably the performance hit from the benchmark

Reply:
- [ ] NewErrorf method